sdk/src/main/scala/com/horizen/account/api/http/AccountEthRpcRoute.scala

```diff
@@ -72,10 +74,29 @@ case class AccountEthRpcRoute(
   def ethRpc: Route = post {
     withAuth {
       entity(as[JsonNode]) { body =>
-        val req = new RpcRequest(body)
+
+        val req = {
+          try {
+            if(body.isArray()) {
+              new RpcBatchRequest(body)}
+            else {
+              new RpcRequest(body)}
+          } catch {
+            case _: RpcException => new RpcResponseError(new RpcId(), RpcError.fromCode(RpcCode.InvalidRequest, null))}
+        }
         log.debug(s"request >> $body")
-        val res = rpcHandler.apply(req)
+
+        val res = {
+          if(req.isInstanceOf[RpcResponseError]) {
+            req}
+          else if(req.isInstanceOf[RpcRequest]) {
+            rpcHandler.apply(req.asInstanceOf[RpcRequest])}
+          else if(req.isInstanceOf[RpcBatchRequest]) {
+            rpcHandler.apply(req.asInstanceOf[RpcBatchRequest])}
+        }
```

Comment:
Let's think more about the code structure here, it seems that we mix things:
1) from one perspective the parse of the body is done here and the knowledge of the the single/batch requests cases is here.
2) from other side, batch requests responses construction are done in `rpcHandler` with a lack of proper parsing of invalid request entries there.
3) error handling is spread between two classes: this one and rpcHandler.

Considering this, I feel there is some room for improvement.
`AccountEthRpcRoute` should be responsible for data parsing and collecting results.
`RpcHandler` should be responsible for single request processing.
In such case we may remove the `BatchRequest` class and leave  `RpcHandler` unchanged

Reply:
My approach aimed to keep the management of input and output creation separate by maintaining the original structure of the `AccountEthRpcRoute` class method, but this led to an incorrect delegation of the error handling between the various components.

Your proposal, which I followed, seems very valid. The `RpcBatchRequest` class has been eliminated and now
the `RpcHandler` is limited to the execution of single rpc requests no longer discriminating between single request and batch. The `AccountEthRpcRoute` takes care of collecting the results instead.